src/component/parser/json/converter/ProcessConverter.ts

```diff
@@ -134,17 +143,9 @@ export default class ProcessConverter {
     this.elementsWithoutParentByProcessId.set(process.id, []);
 
     // flow nodes
-    ShapeUtil.flowNodeKinds()
-      .filter(kind => kind != ShapeBpmnElementKind.EVENT_BOUNDARY)
-      .forEach(kind => this.buildFlowNodeBpmnElements(process[kind], kind, parentId, process.id));
-    // specific management for transaction sub-processes which are handled by the SUB_PROCESS ShapeBpmnElementKind and a dedicated ShapeBpmnSubProcessKind
-    this.buildFlowNodeBpmnElements(process.transaction, ShapeBpmnElementKind.SUB_PROCESS, parentId, process.id, 'transaction');
-    // process boundary events afterward as we need its parent activity to be available when building it
-    this.buildFlowNodeBpmnElements(process.boundaryEvent, ShapeBpmnElementKind.EVENT_BOUNDARY, parentId, process.id);
-
+    orderedFlowNodeBpmnTypes.forEach(bpmnType => this.buildFlowNodeBpmnElements(process[bpmnType], getShapeBpmnElementKind(bpmnType), parentId, process.id, bpmnType));
```

Comment:
That was I had initially mind but this introduces an inconsistency. The name of the method is `buildFlowNodeBpmnElements`. As all similar methods, it take an array of BPMN elements as 1st parameter.
So, this would have required to rename the method, and we wouldn't have seen the similarity with other methods that process BPMN elements (lanes and laneset).

Here is the resulting implementation (without method renaming):
```ts
  private buildFlowNodeBpmnElements(
    process: TProcess | TSubProcess,
    kind: ShapeBpmnElementKind,
    parentId: string,
    processId: string,
    processedSemanticType: BpmnSemanticType,
  ): void {
    ensureIsArray(process[processedSemanticType]).forEach(bpmnElement => {
```

During the implementation, I had no idea for a new method name and I didn't see the gain of changing the method signature comparing to the rest of the changes proposed in this PR.
That's being said, if you see specific arguments, please let me know.

Reply:
After discussion with @csouchet, we decided to keep the current implementation. It could be updated later with the proposal.